app/uk/gov/hmrc/cgtpropertydisposals/models/finance/ClearingReason.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.models.finance
+
+import cats.Eq
+import cats.instances.int._
+import cats.syntax.eq._
+import julienrf.json.derived
+import play.api.libs.json.OFormat
+
+sealed trait ClearingReason
+
+object ClearingReason {
+  case object IncomingPayment extends ClearingReason
+  case object OutgoingPayment extends ClearingReason
+  case object WriteOff extends ClearingReason
+  case object Reversal extends ClearingReason
+  case object MassWriteOff extends ClearingReason
+  case object AutomaticClearing extends ClearingReason
+  case object SomeOtherClearingReason extends ClearingReason
+  case object InvalidClearingReason extends ClearingReason
+
+  def fromString(clearingReason: String): ClearingReason =
```

Comment:
it'd make more sense to have this as:
```scala
def fromString(clearingReason: String): Option[ClearingReason]
```
instead of having `InvalidClearingReason`. That's an error case and shouldn't be passed around to the FE

Reply:
No, because I need to assert against this case in the twirl template to then default to the payment details message. If I error out then I can't asset in the twirl template. See the FE twirl template changes. Or did I not interpret the ticket correctly?

Reply:
the clearing reason being blank was meant to be the case when the clearing reason wasn't given from DES at all, which in the transformer class would be the case:
```scala
 case DesFinancialTransactionItem(Some(paymentAmount), Some(paymentMethod), Some(clearingDate), None) =>
```
Thinking about it now, that means that you could just have 
```scala
case _ => SomeOtherClearingReason
```
at the end of this pattern match and keep it non-optional. Getting an empty string for the field would be weird to get

Reply:
Blank when dealing with strings is not the same as missing as you know. Need to tighten up these story specs. I'll make those changes.